shuffle.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Set configuration
+WORK_DIR="~/work/library_shuffle"
+
+CABLE_REPO="git@github.com:CABLE-LSM/CABLE.git"
+CABLE_BRANCH="538-the-great-library-shuffle"
+CABLE_DIR="{WORK_DIR}/cable"
+CABLE_SHARED_DIR=$CABLE_DIR/src/shared
+CABLE_ESM15_DIR=$CABLE_DIR/src/coupled/ESM1.5
+CABLE_ESM16_DIR=$CABLE_DIR/src/coupled/esm16
+CABLE_ESM_DIR=$CABLE_DIR/src/coupled/esm
+
+UM7_REPO="git@github.com:ACCESS-NRI/UM7.git"
+UM7_BRANCH="64-the-great-library-shuffle"
+UM7_DIR="{WORK_DIR}/um7"
+UM7_ESM15_DIR=$UM7_DIR/umbase_hg3/src/atmosphere/CABLE/src/coupled/ESM1.5
+UM7_CABLE_DIR=$UM7_DIR/umbase_hg3/src/atmosphere/land_surface/cable
+
+# Delete and remake work dir
+rm -rf $WORK_DIR
+mkdir -p $WORK_DIR
+
+# Clone everything
+cd $WORK_DIR
+git clone $CABLE_REPO $CABLE_DIR
+git clone $UM7_REPO $UM7_DIR
+
+# Check out the CABLE branch for the shuffle
+cd $CABLE_DIR
+git fetch origin
+git checkout $CABLE_BRANCH
+
+# Check out specific branches
+cd $UM7_DIR
+git fetch origin
+git checkout $UM7_BRANCH
+
+# Create a src/shared dir in CABLE
+mkdir -p $CABLE_SHARED_DIR
+
+# Move from offline to shared
+mv $CABLE_DIR/src/offline/cable_LUC_EXPT.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/cable_phenology.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/casa_ncdf.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/casa_offline_inout.F90 $CABLE_SHARED_DIR/
+mv $CABLE_DIR/src/offline/cable_pft_params.F90 $CABLE_SHARED_DIR/cable_pft_params_mod.F90
+mv $CABLE_DIR/src/offline/cable_soil_params.F90 $CABLE_SHARED_DIR/cable_soil_params_mod.F90
```

Comment:
I made a mistake in the move. We need to take the files from UM7 (newer) instead of offline:
```suggestion
mv $UM7_ESM15_DIR/cable_pft_params_mod.F90 $CABLE_SHARED_DIR/
mv $UM7_ESM15_DIR/cable_soil_params_mod.F90 $CABLE_SHARED_DIR/

rm $CABLE_DIR/src/offline/cable_pft_params.F90
rm $CABLE_DIR/src/offline/cable_soil_params.F90
```

Reply:
You might want to look at #551 . There are a handful of "modifications" in science/ and uni/ but I haven't looked yet which way they should be going. I suspect they will go back to ESM1.6 and main will;l remain unchanged. Except the one I just looked at. snmin is coming in through a namelist now